polygon/bridge/snapshot_store.go

```diff
@@ -0,0 +1,397 @@
+package bridge
+
+import (
+	"bytes"
+	"context"
+	"encoding/binary"
+	"fmt"
+	"time"
+
+	libcommon "github.com/erigontech/erigon-lib/common"
+	"github.com/erigontech/erigon-lib/common/length"
+	"github.com/erigontech/erigon-lib/downloader/snaptype"
+	"github.com/erigontech/erigon-lib/kv"
+	"github.com/erigontech/erigon-lib/recsplit"
+	"github.com/erigontech/erigon/polygon/heimdall"
+	"github.com/erigontech/erigon/rlp"
+	"github.com/erigontech/erigon/turbo/snapshotsync"
+)
+
+type snapshotStore struct {
+	Store
+	snapshots              *heimdall.RoSnapshots
+	sprintLengthCalculator sprintLengthCalculator
+}
+
+type sprintLengthCalculator interface {
+	CalculateSprintLength(number uint64) uint64
+}
+
+func NewSnapshotStore(base Store, snapshots *heimdall.RoSnapshots, sprintLengthCalculator sprintLengthCalculator) *snapshotStore {
```

Comment:
```suggestion
func NewSnapshotStore(database Store, snapshots *heimdall.RoSnapshots, sprintLengthCalculator sprintLengthCalculator) *snapshotStore {
```

Reply:
Why assume that the base store is a database ?  

I can change it but it seems logically incorrect - if we want to name it after the type shouldn't it just be store ?